palaeontographica-abteilung-b-palaeobotany-palaeophytology.csl

```diff
@@ -0,0 +1,245 @@
+    </terms>
+  </locale>
+  <macro name="anon">
+    <choose>
+      <if variable="author editor translator" match="none">
+        <text term="anonymous" form="short" text-case="capitalize-first"/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="container-contributors">
+    <choose>
+      <if type="chapter paper-conference" match="any">
+        <names variable="editor translator" delimiter=", ">
+          <name name-as-sort-order="all" and="symbol" font-variant="small-caps" initialize-with="." delimiter=", " delimiter-precedes-last="never"/>
+        </names>
+      </if>
+    </choose>
+  </macro>
+  <macro name="author">
+    <names variable="author">
+      <name name-as-sort-order="all" and="symbol" sort-separator=", " initialize-with="." delimiter=", " delimiter-precedes-last="never" font-variant="small-caps"/>
+      <label form="short" prefix=" (" suffix=")"/>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
+      </substitute>
+    </names>
+    <text macro="anon" font-variant="small-caps"/>
+  </macro>
+  <macro name="author-short">
+    <names variable="author">
+      <name form="short" and="symbol" delimiter=", " initialize-with=". " font-variant="small-caps"/>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
+      </substitute>
+    </names>
+    <text macro="anon"/>
+  </macro>
+  <macro name="access">
+    <group delimiter=". " prefix=" ">
+      <choose>
+        <if type="article-journal article-magazine article-newspaper bill book chapter graphic legal_case legislation motion_picture paper-conference report song" match="none">
+          <text variable="container-title" font-style="italic"/>
+        </if>
+      </choose>
+      <choose>
+        <if variable="DOI">
+          <text variable="DOI" prefix="https://doi.org/"/>
+        </if>
+        <else>
+          <choose>
+            <if type="webpage">
+              <group delimiter=" ">
+                <text term="retrieved" text-case="capitalize-first"/>
+                <date variable="accessed" suffix=",">
+                  <date-part name="month" suffix=" "/>
+                  <date-part name="day" suffix=", "/>
+                  <date-part name="year"/>
+                </date>
+                <group delimiter=" ">
+                  <text term="from"/>
+                  <text variable="URL"/>
+                </group>
+              </group>
+            </if>
+          </choose>
+        </else>
+      </choose>
+    </group>
+  </macro>
+  <macro name="publisher">
+    <group delimiter=" ">
+      <text variable="genre" suffix=": " text-case="capitalize-first"/>
+      <choose>
+        <if type="speech paper-conference" match="none">
+          <group delimiter=" ">
+            <text variable="publisher" prefix="(" suffix=")"/>
+            <text variable="publisher-place"/>
+          </group>
+        </if>
+      </choose>
+    </group>
```

Comment:
@damnation333, @adam3smith, the relevant examples at https://www.schweizerbart.de/journals/palb/instructions:

> TAYLOR, T. N. (1981): Paleobotany: An introduction to fossil plant biology. – pp. 1–589 (McGraw-Hill Book Co.) New York.
> 
> FISCHER, O. (1990): Blätterfloren aus mitteleozänen Sedimenten des südlichen Weißelster-Beckens (Profen und Scheiplitz). – pp. 1–142, unpublished PhD thesis, Humboldt University Berlin.

You would store the "unpublished PhD thesis" in the "genre" CSL variable, right? In that case this doesn't look good yet.